### DISCUSSION TITLE
Do I miss something not having specific compilation flags on ARM64 platform?

### DISCUSSION THREAD

**Discussion by gotzmann** [ggml-org/llama.cpp#1723] [General]
I'm running llama.cpp on Ampere Altra Q80 server and all is fine there, but the performance with 16 threads just not so good as on my sweetie M1 Pro laptop with just 6 threads. So I've started thinking maybe I'm missing something and some explicit compile time flags might help there? Maybe some hardware optimisations just not used with default `make`?

**Comment by ghost** [comment]
> I'm running llama.cpp on Ampere Altra Q80 server and all is fine there, but the performance with 16 threads just not so good as on my sweetie M1 Pro laptop with just 6 threads. So I've started thinking maybe I'm missing something and some explicit compile time flags might help there? Maybe some hardware optimisations just not used with default `make`?

Hi,

Increased --threads # does not necessarily equal better performance. It depends on the device/system, though it's also possible to build llama.cpp without certain libraries, for example OpenBlas, CLBlast, etc.

Without make LLAMA_OPENBLAS=1, or make LLAMA_CLBLAST=1 then llama.cpp will not include the libraries.
